scheduler/filter/slots.go

```diff
@@ -0,0 +1,48 @@
+package filter
+
+import (
+	"errors"
+	"github.com/docker/swarm/cluster"
+	"github.com/docker/swarm/scheduler/node"
+	"strconv"
+)
+
+var (
+	// ErrNoNodeWithFreeSlotsAvailable is exported
+	ErrNoNodeWithFreeSlotsAvailable = errors.New("No node with enough open slots available in the cluster")
+)
+
+//SlotsFilter only schedules containers with open slots.
+type SlotsFilter struct {
+}
+
+// Name returns the name of the filter
+func (f *SlotsFilter) Name() string {
+	return "containerslots"
+}
+
+// Filter is exported
+func (f *SlotsFilter) Filter(_ *cluster.ContainerConfig, nodes []*node.Node, _ bool) ([]*node.Node, error) {
+	result := []*node.Node{}
+
+	for _, node := range nodes {
+
+		if slotsString, ok := node.Labels["containerslots"]; ok {
+			slots, err := strconv.Atoi(slotsString) //if err => cannot cast to int, so no candidate for us
```

Comment:
> what about not using a label, but a command line argument for the swarm agent that has to be int?

If you want to add this to Docker Daemon, I think label is the only option. You can control each docker engine's slot number. The other option is to add it to Swarm manager. The problem is that it'd apply to all docker engines equally, i.e., every node can host *k* containers. It's not good if some nodes have more resources or they run different jobs. 

> what about the first suggestion, to fail but log something, so if the user wonders why his containers won't start at node xy he looks into the log and sees what the reason is?

Again, logs are not obvious as user may not look at it. So I consider this as failing implicitly. My suggestion is to ignore an *invalid* label and let container run. It's consistent with how we deal with other labels.